provider/source/src/units/ids.rs

```diff
@@ -0,0 +1,74 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use std::collections::BTreeMap;
+use std::collections::HashSet;
+
+use icu::experimental::measure::provider::ids::Index;
+use icu::experimental::measure::provider::ids::UnitsIdsV1;
+use icu_provider::prelude::*;
+use icu_provider::DataError;
+use icu_provider::DataMarkerAttributes;
+use icu_provider::DataProvider;
+use icu_provider::DataRequest;
+use icu_provider::DataResponse;
+
+use crate::cldr_serde;
+use crate::SourceDataProvider;
+
+impl DataProvider<UnitsIdsV1> for SourceDataProvider {
+    fn load(&self, req: DataRequest) -> Result<DataResponse<UnitsIdsV1>, DataError> {
+        self.check_req::<UnitsIdsV1>(req)?;
+        let units_data: &cldr_serde::units::info::Resource = self
+            .cldr()?
+            .core()
+            .read_and_parse("supplemental/units.json")?;
+
+        let unit = req.id.marker_attributes.as_str();
+
+        let indices_map: BTreeMap<String, usize> = units_data
+            .supplemental
+            .convert_units
+            .convert_units
+            .keys()
+            .enumerate()
+            .map(|(index, unit_name)| (unit_name.clone(), index))
+            .collect();
+
+        let index = indices_map
```

Comment:
it's probably faster to use `.find` on the iterator than to create a map for a single lookup